### ISSUE TITLE
Setting favicon breaks loginSuccessUrl in SpringBoot app

### ISSUE THREAD

**Issue by mkpaz** [vaadin/flow#17342]
### Description of the bug

Setting favicon via `addFavIcon()` methods breaks the login flow. After form submit app redirects to the `http://<whatever>/icons/favicon.png`. This is because `VaadinDefaultRequestCache` stores the previous request. While setting ...

```java
http.requestCache(c -> c.requestCache(new NullRequestCache()));
```

.. solves the problem, I think Vaadin request cache should be configured accordingly.

PS: I tried _every_ favicon path specified in the doc (including `favicon.icon` which I believe is a typo, because such extension doesn't exist) and none of them working, which I think is another problem.

### Expected behavior

Exclude favicons from security request cache.

### Minimal reproducible example

```
@SpringBootApplication
public class Application implements AppShellConfigurator {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Override
    public void configurePage(AppShellSettings settings) {
        settings.addFavIcon("icon", "icons/favicon.png", "64x64");
    }
}

@EnableWebSecurity
@Configuration
public class SecurityConfiguration extends VaadinWebSecurity {

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http.authorizeHttpRequests(auth ->
            auth.requestMatchers(new AntPathRequestMatcher("/public/**")).permitAll()
        );

        super.configure(http);

        setLoginView(http, LoginPage.class);
    }

    @Override
    public void configure(WebSecurity web) throws Exception {
        super.configure(web);
    }

    @Bean
    public UserDetailsManager userDetailsService() {
        UserDetails admin = User.withUsername("admin")
            .password("{noop}admin")
            .roles("ADMIN")
            .build();
        return new InMemoryUserDetailsManager(admin);
    }
}
```

I only added favicon method. Everything else is from [tutorial](https://vaadin.com/docs/latest/security/enabling-security).

### Versions

- Vaadin / Flow version: 24.1.4
- Java version: 17
- OS version: Fedora 37
- Spring Boot version: 3.1.2

**Comment by mcollovati** [comment]
@mkpaz thanks for the issue.
Custom favicon should be handled by the framework and made publicly accessible by default.
As a workaround, you can add a `permitAll()` request matcher for  the favicon path

```java
        http.authorizeHttpRequests(auth ->
                .requestMatchers(
                        new AntPathRequestMatcher("/public/**"),
                        new AntPathRequestMatcher("/icons/favicon.png")
               ).permitAll()
        );
```

**Comment by sveinnetnordic** [comment]
Same issue for `@PWA(iconPath = "core/img/studd-small.png")`
Could use nearly the same workaround: 
`http.authorizeHttpRequests { authorize -> authorize.requestMatchers(AntPathRequestMatcher("/core/img/studd-small-*.png")).permitAll() }`

**Comment by mshabarov** [comment]
@mkpaz the fix is ready and going to be included into next Vaadin 24.2 and 24.3 maintenance release. The Vaadin 24.1 is out of maintenance currently, so we aren't going to release it anymore. Please upgrade to 24.2 or 24.3 to have this fixed.
